src/app/build.gradle

```diff
@@ -0,0 +1,85 @@
+apply plugin: 'com.android.application'
+apply plugin: 'android-apt'
+apply from: '../jacoco.gradle'
+
+android {
+    compileSdkVersion 23
+    buildToolsVersion "23.0.3"
+
+    defaultConfig {
+        minSdkVersion 14
+        targetSdkVersion 23
+        applicationId "br.com.italoromeiro.arctouchtest"
+        testApplicationId 'br.com.italoromeiro.arctouchtest.test'
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        versionCode 1
+        versionName "1.0"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            debuggable true
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+
+        debug {
+            applicationIdSuffix '.dev'
+            testCoverageEnabled true
+            ext.betaDistributionGroupAliases = "android-testers"
+        }
+    }
+
+    testOptions {
+        resultsDir = "${project.buildDir}/foo/results"
+    }
+}
+
+apt {
+    arguments {
+        resourcePackageName android.defaultConfig.applicationId
+        androidManifestFile variant.outputs[0]?.processResources?.manifestFile
+    }
+}
+
+configurations.all {
+    resolutionStrategy.force 'com.android.support:support-annotations:23.4.0'
+    resolutionStrategy.force 'com.android.support:appcompat-v7:23.4.0'
+    resolutionStrategy.force 'com.android.support:design:23.4.0'
+    resolutionStrategy.force 'com.android.support:recyclerview-v7:23.4.0'
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+
+    // UNIT TESTS
+    testCompile 'junit:junit:4.12'
+    testCompile 'org.hamcrest:hamcrest-library:1.3'
+    testCompile 'org.mockito:mockito-core:2.0.100-beta'
+
+    // ANDROID TESTS
+    androidTestCompile 'com.android.support.test:runner:0.5'
+    androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
+    androidTestCompile 'com.android.support.test.espresso:espresso-contrib:2.2.2'
+    androidTestCompile 'com.squareup.okhttp3:mockwebserver:3.4.1'
+    androidTestCompile 'com.squareup.retrofit2:retrofit-mock:2.1.0'
+
+    // GOOGLE APIs
+    compile 'com.android.support:appcompat-v7:23.4.0'
+    compile 'com.android.support:design:23.4.0'
+    compile 'com.android.support:recyclerview-v7:23.4.0'
+    compile 'com.google.android.gms:play-services-location:9.4.0'
+    compile 'com.google.android.gms:play-services-maps:9.4.0'
+
+    // ANDROID ANNOTATIONS
```

Comment:
From my point of view I can say that:

Pros:
- You don't need to write boilerplate code, so you gain time to do a better job;
- The amount of annotations provided by Android Annotations is quite good, a lot of effort from you can be saved when you work with it;
- The injection of code occurs at compile time, so at runtime you don't have a lot of heavy operations going on at once in the background;

Cons:
- All the annotated classes need to gain a Class_. These generated classes inject the boilerplate code and, because of that, you need to deal with they. It's a bad thing, because you need to change the use of the natural classes to these "weird" classes both in manifest and in the current classes;
- Proguard is another problem you need to deal with. When you are in the release mode, for instance, if you forget to configure your proguard, certainly you will have trouble because the automatic process used by proguard can't infer that there are certain special uses of the annotations and some reflection can crash;
- The "new" operation is a costly operation. So, if you think you have X classes, all of them annotated, actually you have 2X "new" operations to do. So, we can see it is a loss of performance for the app.

So, with some special cares, I think it is a good idea to use tools like that. The mobile devices today have a lot of hardware to support these kind of situations and, with some experience that is natural with the time, the gain of coding time is incredible.